org.lamport.tla.toolbox.tool.tlc/src/org/lamport/tla/toolbox/tool/tlc/job/TLCProcessJob.java

```diff
@@ -238,9 +249,19 @@ protected IStatus run(IProgressMonitor monitor)
 
                 listener = new BroadcastStreamListener(model, kind);
 
-                process.getStreamsProxy().getOutputStreamMonitor().addListener(listener);
-                process.getStreamsProxy().getErrorStreamMonitor().addListener(listener);
-
+                // stdout, stderr, stdin of the forked TLC process.
+                final IStreamsProxy streamsProxy = process.getStreamsProxy();
+				streamsProxy.getOutputStreamMonitor().addListener(listener);
+				streamsProxy.getErrorStreamMonitor().addListener(listener);
+				
+				// This is a reader wrapping the input stream of the Toolbox console. In other
+				// words, if a user types a string into the Toolbox console, we can read the
+				// string from the reader. Iff consoleStdIn is not null (there is a Toolbox
+				// console), the loop below reads from consoleStdIn and writes (forwards) the
+				// string to stdin of the TLC process.  Because the while loop sleeps for a
+				// second, the response time isn't perfect but is good enough for now.
+				final BufferedReader consoleReader = getConsoleReader();

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *consoleReader*. Currently, there are execution paths that do not contain closure statements. Either a) close *consoleReader* in a try-finally block or b) close the resource by declaring *consoleReader* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).